### PR TITLE
[WIP]Allow bots to upload files.

### LINKED ISSUE #351: Add file upload access to the bot API
A full-fledged Client python object has [```upload_file()```](https://github.com/zulip/python-zulip-api/blob/master/zulip/zulip/__init__.py#L665), but [a bot's API exposed via bot_handler](https://zulipchat.com/api/writing-bots#bot-api) doesn't currently include that method. 

While there are non-ideal ways to access that method from a bot's ```handle_message```, it would be nice to have it properly promoted up to the public ```bot_handler``` API. Could we add it?

Uploading files is handy for bots in charge of generating and conveying back imagery or other types of small files to the caller.

### PR DESCRIPTION
**Pull request by SivagiriVisakan**
This commit exposes the already existing method in
Zulip Client to upload files.

Fixes #351

### CODE REVIEW

**Comment by timabbott** [comment]
You should add a test for this.

File: zulip_bots/zulip_bots/lib.py [hunk 1]
```diff
@@ -158,6 +158,24 @@ def update_message(self, message):
...
+            return self._client.upload_file(file)
+        else:
+            self._rate_limit.show_error_and_exit()
+
+    def upload_file_from_path(self, file_path):
+        #type: (str) -> Dict[str, Any]
+        try:
+            abs_filepath = os.path.abspath(file_path)
+            file = open(abs_filepath,'rb')
+            return self.upload_file(file)
```
[the remark this answers is not in the document]
Comment (SivagiriVisakan) [reply zulip_bots/zulip_bots/lib.py]: Thanks for catching that !
I have modified it as per your suggestion.

**Comment by SivagiriVisakan** [comment]
Example usage (to upload an image): 

```python
try:
    upload = bot_handler.upload_file_from_path('image.jpg')
    # `path/to/image` is realtive to the directory from where the bot is run in terminal.
    # For example , if the `zulip-run-bot` command is run from a directory `home/mydirectory`, 
    # then the bot will try to upload the file at `home/mydirectory/image.jpg`

    # upload will be the following dict if file upload succeeded
    # {'msg': '', 'result': 'success', 
    #   'uri': '/user_uploads/2/d0/C1AudhaEiLNxH7yqgN4A0nh0/image.jpg'}

    image_uploaded_reply = '[Uploaded Image]({})'.format(upload['uri'])
    bot_handler.send_reply(message, image_uploaded_reply)
except OSError:  # Raised on any error while reading the file at the path
    pass
```

The above message looks like this : 
bot-file-upload

File: zulip_bots/zulip_bots/lib.py [hunk 2]
```diff
@@ -158,6 +158,33 @@ def update_message(self, message):
         else:
             self._rate_limit.show_error_and_exit()
 
+    def upload_file(self, file):
+        # type: (IO[Any]) -> Dict[str, Any]
```
Comment (noemartinez) [inline zulip_bots/zulip_bots/lib.py]: I believe this line should change to match the new return type.

File: zulip_bots/zulip_bots/lib.py [hunk 3]
```diff
@@ -158,6 +158,33 @@ def update_message(self, message):
...
+                else:
+                    error_msg = upload['msg']
+                    raise FileUploadError(error_msg, upload)
+            except KeyError:
+                raise FileUploadError('Error: Received unexpected payload', upload)
+        else:
+            self._rate_limit.show_error_and_exit()
+
+    def upload_file_from_path(self, file_path):
+        # type: (str) -> Dict[str, Any]
```
Comment (noemartinez) [inline zulip_bots/zulip_bots/lib.py]: I believe this line should change to match the new return type.

**Comment by novokrest** [comment]
@SivagiriVisakan  Would you like to work more on this?

**Comment by timabbott** [comment]
@novokrest I think given how long it's been since the review comments, we should just close this and you can take it over.

**Pull request closed without merging**